### ISSUE TITLE
Not able to install vagran-vbguest in macox 

### ISSUE THREAD

**Issue by byteshiva** [dotless-de/vagrant-vbguest#186]
Not able to install vagrant-vbguest in macox 

```
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that

the guest additions are properly installed in the guest and

can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` home_vagrant_Mod /home/vagrant/Mod

mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` home_vagrant_Mod /home/vagrant/Mod

The error output from the last command was:

mount: unknown filesystem type 'vboxsf' 
```

**Comment by fnordfish** [comment]
Hi @byteshiva,

I'm sorry, but your report is a bit confusing. It shows that virtual box guest additions are not working/installed inside the guest os. I cannot see, how the installation of the vagrant-vbguest plugin failed though. So, I'd need some more details to see what might be wrong:

Which version of vagrant (`vagrant -v`) and vagrant-vbguest (`vagrant plugin list`) are you using?
If `vagrant plugin list` does not show the line `vagrant-vbguest (0.11.0)`, please install the newest version with `vagrant plugin install vagrant-vbguest`.

I've figured that you are using Mac OS X as a host system, what specific guest system are you using?
Can you paste the output of `vagrant up` and/or `vagrant vbguest` into a gist and link it here please.

**Comment by byteshiva** [comment]
```
$vagrant -v
Vagrant 1.7.4

$vagrant plugin list
vagrant-rsync (0.2.5)
vagrant-share (1.1.5, system)
vagrant-triggers (0.5.2)
vagrant-vbguest (0.11.0)
vagrant-vbox-snapshot (0.0.10)
```
I'm using centos7 as the guest system. I'm just trying to do simple task like port forwarding and rsync folders from HOST to Guest Machine.

**Comment by fnordfish** [comment]
AFAIK, for neither would you *need* guest additions. Maybe I'm wrong with the port forwarding, but rsync should definitely work.

**Comment by byteshiva** [comment]
@fnordfish without guest addon, I keep getting this error saying no guest add-on detected. It throws error saying it needs guest addon to continue port forwarding. I will include rather detailed log for your reference.

**Comment by kmugrage** [comment]
I had about 20 people in a workshop today where I provided a box and some other files on a USB drive. They copied the files over to their systems and then ran 'vagrant up'. 3 of the attendees got the error in this issue. 

After investigation, we learned that vagrant-vbguest was installed by default on their company load, and it was trying to update the guest. The image I handed out didn't have kernel headers, so the new guest additions failed to build. 

If they removed vagrant-vbguest it worked or installed the kernel headers it would work.

I'll add the headers to the box so it doesn't happen again, but putting this here in case others have the same issue.

Not sure if you'd consider this a feature request, but It seems like it should be able to check if the addition of new guest additions would succeed before removing the old ones.
